### DISCUSSION TITLE
Converting a disc duplicator to a disc ripper.

### DISCUSSION THREAD

**Discussion by birdman3131** [automatic-ripping-machine/automatic-ripping-machine#828] [General]
So I have an Microboards Orbit 3. It has a 100 disc hopper and can load and unload discs automatically.

Current thought is to replace the PATA drives with Blu-ray sata drives and the motor controller with some flavor of ESP/Arduino board.

ARM looks like it is 90% of the way to doing what I need. The issue is that I need it to handle the disc eject/load sequnce as well. Currently looking like it would be serial commands over usb to a microcontroller and wait for a response.

https://youtu.be/yxKq2pM7RLM?t=95 shows a bit of the mechanicals.

The power on clearing sequence would be something along the lines of this.

1. Lower arm till you hit lower limit.
2. Eject drive 1.
3. Raise arm till upper limit hit.
4. Close drive.
5. Lower arm till lower limit.
6. Send disc forward/backward out of the machine. (Foward for succesful rips. Backward for failed rips.)
7. Repeat 2-6 for drive 2

Then to rip a disc.

1. Eject drive
2. Raise arm till you hit upper limit.
3. Cycle disc drop mechinism.
4. Lower arm.
5. Close drive.
6. Rip disc. (Let ARM do its thing)
7. Disc eject sequence. (2-6 above)
8. Take photo of disc. (For those discs with no meta data so AI or you can name it later.)

Is this doable with ARM? It sounds almost like it is but not positive. (Camera would be a nice to have but not totally needed for most people.)

**Comment by microtechno9000** [comment]
This is quite a specific thing for ARM to incorporate; however, it could be possible through executing a script on job completion. I think scripts on completion are already supported, not certain as I haven't tested or used them. Have a look through the ARM config.

**Reply by birdman3131** [reply]
True. Which the above long sequence was me copying my reddit post on the machine. I assume much of that would be handled by my microcontroller.

In reality all the fancy stuff should be doable with a script on ending like you said. The script would load a new disc and arm would just take off. Only question would be if I could determine a failure. (Assuming scripts work.)

**Reply by turnkit** [reply]
Are the Orbis and the Primera the most popular duplicators?  Supporting both would cover a lot of needs.  

The "stack 100 discs and rip" idea is really helpful for a lot of us.
